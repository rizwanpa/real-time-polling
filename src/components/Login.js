import React, { Component } from "react";
import { BrowserRouter as Router,Redirect } from "react-router-dom";
import { connect } from "react-redux";
import { Row, Col, Form, Input, Button, Alert,message } from "antd";
import { history } from "../store";
import { setSecurity } from "./../actions/userAction";
import {MESSAGE} from './../constants/dictionary';


class Login extends Component {
  state = {
    errorMessage : ''
  };
  componentDidMount() {
    let accessToken = sessionStorage.getItem("accessToken");
    console.log("accessToken000", accessToken, accessToken !== null,accessToken !== "" && accessToken !== null && accessToken !== undefined);
    if (accessToken !== "" && accessToken !== null && accessToken !== undefined) {
       //this.props.history.push("/");
    }
  }
  /* componentWillReceiveProps(nextProps) {
    // if(nextProps.user.name)
    //     this.props.history.push("/");
    // else
    //     this.props.history.push("/login");
  } */
  componentDidUpdate(prevProps) {
    console.log("componentsDidUpdate prevProps--->", prevProps);
    let { username } = this.props.user !== undefined ? this.props.user : "";
    let { id } = this.props.user !== undefined ? this.props.user : "";
    let accessToken = sessionStorage.getItem("accessToken");
    console.log("componentDidUpdate ==> accessToken000", accessToken);
    if (!accessToken) {
      message.error(MESSAGE.authError);
       //this.props.history.push("/login");
    }
  }
  onFinishFailed = errorInfo => {};
  onFinish = values => {
    this.props.setSecurity(values);
  };
  render() {
    const layout = {
      labelCol: {
        span: 8
      },
      wrapperCol: {
        span: 8
      }
    };
    const tailLayout = {
      wrapperCol: {
        offset: 8,
        span: 8
      }
    };
    let accessToken = sessionStorage.getItem("accessToken");
    if (accessToken) {
      return <Redirect to='/' />
    }
    return (
      <Router history={history}>
        <div className='login-background'>
          <section className="login">
          {this.state.errorMessage !=='' ?
          <Alert message="Error Text" style={{top: '-15px'}} type="error" /> :null
        }
            <Form
              {...layout}
              name="LoginForm"
              id="LoginForm"
              initialValues={{
                remember: true
              }}
              onFinish={this.onFinish}
              onFinishFailed={this.onFinishFailed}
            >
              <Form.Item
                label={<label style={{ color: "#fff" }}>Email</label>}
                name="email"
                rules={[
                  {
                    required: true,
                    message: "Please input your email!"
                  }
                ]}
              >
                <Input />
              </Form.Item>

              <Form.Item
                label={<label style={{ color: "#fff" }}>Password</label>}
                name="password"
                rules={[
                  {
                    required: true,
                    message: "Please input your password!"
                  }
                ]}
              >
                <Input.Password />
              </Form.Item>

              {/* <Form.Item {...tailLayout} name="remember" valuePropName="checked">
              <Checkbox>Remember me</Checkbox>
            </Form.Item> */}

              <Form.Item {...tailLayout}>
                <Button type="primary" htmlType="submit">
                  Login
                </Button>
              </Form.Item>
            </Form>
          </section>
        </div>
      </Router>
    );
  }
}

const mapStateToProps = state => {
  return { user: state.user };
};
const mapDispatchToProps = {
  setSecurity
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
