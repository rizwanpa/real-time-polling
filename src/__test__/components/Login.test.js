import React from "react";
import { Provider } from "react-redux";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store"; // Smart components

// Component to be tested
import Login from "./../../components/Login";
configure({ adapter: new Adapter() });
// create any initial state needed
const initialState = {
  username: null,
  id: "",
  errors: {}
};
// here it is possible to pass in any middleware if needed into //configureStore
const mockStore = configureStore();
const store = mockStore({
  username: null,
  id: "",
  errors: {}
});

describe("Login Component", () => {
  let wrapperLogin;
  // our mock login function to replace the one provided by mapDispatchToProps
  const mockLoginfn = jest.fn();
  describe("render()", () => {
    test("renders the component", () => {
      const wrapper = shallow(
        <Provider store={store}>
          <Login />
        </Provider>
      );
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
