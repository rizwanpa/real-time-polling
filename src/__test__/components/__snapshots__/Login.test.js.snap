// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component render() renders the component 1`] = `<Connect(Login) />`;
